### PR TITLE
Create /etc/systemd/network dir for ifname= renaming if needed.

### PR DESCRIPTION
**Pull request by rvykydal**
Resolves: rhbz#1841469

### CODE REVIEW

**Comment by jstodola** [comment]
Tested using updates.img - this change fixes the traceback and the network device is correctly renamed on the installed system.

File: pyanaconda/modules/network/installation.py [hunk 1]
```diff
@@ -179,16 +180,23 @@ def _write_interface_rename_config(self, root, ifname_option_values, overwrite):
         :param overwrite: overwrite existing configuration file
         :type overwrite: bool
         """
+
+        if ifname_option_values:
+            target_system_dir = os.path.normpath(root + self.SYSTEMD_NETWORK_CONFIG_DIR)
```
Comment (poncovka) [inline pyanaconda/modules/network/installation.py:185]: We try to use `join_paths` now, but it is up to you.

File: pyanaconda/modules/network/installation.py [hunk 2]
```diff
@@ -179,16 +180,23 @@ def _write_interface_rename_config(self, root, ifname_option_values, overwrite):
...
+            if not os.path.exists(target_system_dir):
+                util.mkdirChain(target_system_dir)
+
         for ifname_value in ifname_option_values:
             iface, mac = ifname_value.split(":", 1)
             content = self.INTERFACE_RENAME_FILE_CONTENT_TEMPLATE.format(mac, iface)
-            file_path = self.INTERFACE_RENAME_FILE_TEMPLATE.format(iface)
+            config_file = self.INTERFACE_RENAME_FILE_TEMPLATE.format(iface)
+            config_file_path = os.path.join(self.SYSTEMD_NETWORK_CONFIG_DIR, config_file)
```
Comment (poncovka) [inline pyanaconda/modules/network/installation.py:193]: Same here.

File: pyanaconda/modules/network/installation.py [hunk 3]
```diff
@@ -179,16 +180,23 @@ def _write_interface_rename_config(self, root, ifname_option_values, overwrite):
         :param overwrite: overwrite existing configuration file
         :type overwrite: bool
         """
+
+        if ifname_option_values:
+            target_system_dir = os.path.normpath(root + self.SYSTEMD_NETWORK_CONFIG_DIR)
+            if not os.path.exists(target_system_dir):
```
Comment (poncovka) [inline pyanaconda/modules/network/installation.py:186]: You don't have to check it. The function `mkdirChain` shouldn't fail if the directories exist.

File: pyanaconda/modules/network/installation.py (hunk 1 above)
Comment (rvykydal) [reply pyanaconda/modules/network/installation.py]: Used.

File: pyanaconda/modules/network/installation.py (hunk 2 above)
Comment (rvykydal) [reply pyanaconda/modules/network/installation.py]: Used.

File: pyanaconda/modules/network/installation.py (hunk 3 above)
Comment (rvykydal) [reply pyanaconda/modules/network/installation.py]: Removed the check.

**Comment by rvykydal** [comment]
Well, I need to update the tests

**Comment by rvykydal** [comment]
Ported to master with https://github.com/rhinstaller/anaconda/pull/2626

**Pull request merged**